models/User.ts

```diff
@@ -0,0 +1,68 @@
+import { model, Schema } from "mongoose";
+import IUser from "../types/IUser";
+
+const UserSchema = new Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  givenName: {
+    type: String,
+    required: true,
+  },
+  familyName: {
+    type: String,
+    required: true,
+  },
+  email: {
+    type: String,
+    required: true,
+  },
+  profilePicture: {
+    type: String,
+    required: true,
+  },
+  bio: {
+    type: String,
+    required: false,
+  },
+  winStreak: {
+    type: Number,
+    default: 0,
+  },
+  longestWinStreak: {
+    type: Number,
+    default: 0,
+  },
+  lastWin: {
+    type: Date,
+    default: null,
+  },
+  gamesPlayed: {
+    type: Number,
+    default: 0,
+  },
+  gamesWon: {
+    type: Number,
+    default: 0,
+  },
+  guessDistribution: {
+    type: [Number],
+    default: [0, 0, 0, 0, 0, 0],
+  },
+  pastGuesses: {
+    type: [[[Number]]],
```

Comment:
Would each of the outermost lists in this 3D array represent the guesses for a different puzzle? It might be a good idea to clarify this with a comment somewhere, since 3D arrays can be hard to decipher without some sort of explanation.

Reply:
You are totally right, this definitely needs a clarification comment